oshmem/mca/atomic/ucx/atomic_ucx_module.c

```diff
@@ -32,6 +32,84 @@ int mca_atomic_ucx_finalize(void)
     return OSHMEM_SUCCESS;
 }
 
+static inline
+int mca_atomic_ucx_op(void *target,
+                      uint64_t value,
+                      size_t size,
+                      int pe,
+                      ucp_atomic_post_op_t op)
+{
+    ucs_status_t status;
+    ucs_status_ptr_t status_ptr;
+    spml_ucx_mkey_t *ucx_mkey;
+    uint64_t rva;
+    uint64_t val;
+
+    if ((8 != size) && (4 != size)) {
+        ATOMIC_ERROR("[#%d] Type size must be 4 or 8 bytes.", my_pe);
+        return OSHMEM_ERROR;
+    }
+
+    ucx_mkey = mca_spml_ucx_get_mkey(pe, target, (void *)&rva);
+    status = ucp_atomic_post(mca_spml_self->ucp_peers[pe].ucp_conn,
+                             op, value, size, rva,
+                             ucx_mkey->rkey);
+    return ucx_status_to_oshmem(status);
+}
+
+static inline
+int mca_atomic_ucx_fop(void *target,
+                       void *prev,
+                       uint64_t value,
+                       size_t size,
+                       int pe,
+                       ucp_atomic_fetch_op_t op)
+{
+    ucs_status_t status;
+    ucs_status_ptr_t status_ptr;
+    spml_ucx_mkey_t *ucx_mkey;
+    uint64_t rva;
+
+    if ((8 != size) && (4 != size)) {
```

Comment:
can we do this as assertion (to avoid in release mode)?

Reply:
done